foreman/data_refinery_foreman/gatherer/management/commands/gather_accessions.py

```diff
@@ -217,28 +201,52 @@ def handle(self, *args, **options):
                 continue
             agents.append(cls(options))
 
-        entries = set()
+        gathered_accessions = set()
         for agent in agents:
-            entries.update(agent.collect_data())
+            agent_accessions = agent.collect_data()
+            agent_accessions_count = len(agent_accessions)
+
+            gathered_accessions.update(agent_accessions)
 
-        entries = sorted(  # Sort the resulting list.
-            (entry for entry in entries if self.RE_ACCESSION.match(entry.accession_code)),
-            key=lambda entry: (
-                self.RE_ACCESSION.match(entry.accession_code).group(1),
-                int(self.RE_ACCESSION.match(entry.accession_code).group(2)),
+            logger.info(
+                f"Since {options['since']} {agent} has gathered {agent_accessions_count} "
+                f"new accession{pluralize(agent_accessions_count)}."
+            )
+
+        gathered_accessions = sorted(  # Sort the resulting list.
+            (ga for ga in gathered_accessions if self.RE_ACCESSION.match(ga.accession_code)),
+            key=lambda ga: (
+                self.RE_ACCESSION.match(ga.accession_code).group(1),
+                int(self.RE_ACCESSION.match(ga.accession_code).group(2)),
             ),
         )
         # Limit the number of output entries.
-        entries = entries[: options["count"]] if options["count"] else entries
+        gathered_accessions = (
+            gathered_accessions[: options["count"]] if options["count"] else gathered_accessions
+        )
+
+        agents_count = len(agents)
+        gathered_accessions_count = len(gathered_accessions)
+        logger.info(
+            f"Since {options['since']} {gathered_accessions_count} "
+            f"new accession{pluralize(gathered_accessions_count)} "
+            f"h{pluralize(gathered_accessions_count, 'as,ave')} been gathered by {agents_count} "
+            f"accession agent{pluralize(agents_count)}."
+        )
 
         if options["dry_run"]:
-            if entries:
-                output = "\n".join((str(entry) for entry in entries))
+            if gathered_accessions:
+                output = "\n".join((str(ga) for ga in gathered_accessions))
             else:
-                output = "No accessions found."
+                output = "No new accessions gathered."
             print(output)
         else:
             try:
-                GatheredAccession.objects.bulk_create(entries)
+                GatheredAccession.objects.bulk_create(gathered_accessions)
```

Comment:
I think we should instead do the `bulk_create` in batches and log each successful / unsuccessful batch.

Reply:
Done.